lib/vehicle.rb

```diff
@@ -1,25 +1,24 @@
 require 'date'
 
 class Vehicle
-  attr_reader :vin,
-              :year,
-              :make,
-              :model,
-              :engine
+  attr_accessor :vehicle_details, :vin, :year, :make, :model, :engine, :registration_date, :plate_type
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = vehicle_details[:registration_date]
+    @plate_type = vehicle_details[:plate_type]
   end
 
   def antique?
-    Date.today.year - @year > 25
+    Date.today.year - @year >= 25
   end
 
   def electric_vehicle?
     @engine == :ev
   end
+
 end
```

Comment:
We really only want to create attr_accessor's for variables if we want those variables to be able to get updated outside of the class. The only attributes that I think might get updated outside of this class are the `@registration_date` and `@plate_type`, so those will be the only attr_accessor's we'll need. The others can be attr_readers. That said, I don't think you have an attribute called `vehicle_details`, so you could get rid of that (it would probably return `nil` at this moment if it were to get called.